core/src/test/java/com/airbnb/aerosolve/core/models/LowRankLinearModelTest.java

```diff
@@ -0,0 +1,265 @@
+  }
+
+  Map<String, FloatVector> makeLabelWeightVector() {
+    Map<String, FloatVector> labelWeights = new HashMap<>();
+    float[] animalFeature = {1.0f, 0.0f, 0.0f};
+    float[] colorFeature = {0.0f, 1.0f, 0.0f};
+    float[] fruitFeature = {0.0f, 0.0f, 1.0f};
+    labelWeights.put("A", new FloatVector(animalFeature));
+    labelWeights.put("C", new FloatVector(colorFeature));
+    labelWeights.put("F", new FloatVector(fruitFeature));
+    return labelWeights;
+  }
+
+  LowRankLinearModel makeLowRankLinearModel() {
+    // A naive model with three classes 'animal', 'color' and 'fruit'
+    // and the size of embedding D = number of labels, W is an identity matrix
+    LowRankLinearModel model = new LowRankLinearModel();
+    model.setEmbeddingDimension(3);
+    model.setLabelDictionary(makeLabelDictionary());
+
+    // construct featureWeightVector
+    Map<String, Map<String, FloatVector>> featureWeights = new HashMap<>();
+    Map<String, FloatVector> animalFeatures = new HashMap<>();
+    Map<String, FloatVector> colorFeatures = new HashMap<>();
+    Map<String, FloatVector> fruitFeatures = new HashMap<>();
+
+    String[] animalWords = {"cat", "dog", "horse", "fish"};
+    String[] colorWords = {"red", "black", "blue", "white", "yellow"};
+    String[] fruitWords = {"apple", "kiwi", "pear", "peach"};
+    float[] animalFeature = {1.0f, 0.0f, 0.0f};
+    float[] colorFeature = {0.0f, 1.0f, 0.0f};
+    float[] fruitFeature = {0.0f, 0.0f, 1.0f};
+
+    for (String word: animalWords) {
+      animalFeatures.put(word, new FloatVector(animalFeature));
+    }
+
+    for (String word: colorWords) {
+      colorFeatures.put(word, new FloatVector(colorFeature));
+    }
+
+    for (String word: fruitWords) {
+      fruitFeatures.put(word, new FloatVector(fruitFeature));
+    }
+    featureWeights.put("a", animalFeatures);
+    featureWeights.put("c", colorFeatures);
+    featureWeights.put("f", fruitFeatures);
+    model.setFeatureWeightVector(featureWeights);
+
+    // set labelWeightVector
+    model.setLabelWeightVector(makeLabelWeightVector());
+    model.buildLabelToIndex();
+
+    return model;
+  }
+
+  public FeatureVector makeFeatureVector(String label) {
+    FeatureVector featureVector = new FeatureVector();
+    HashMap stringFeatures = new HashMap<String, HashSet<String>>();
+    featureVector.setStringFeatures(stringFeatures);
+    HashMap floatFeatures = new HashMap<String, HashMap<String, Double>>();
+    featureVector.setFloatFeatures(floatFeatures);
+    HashMap feature = new HashMap<String, Float>();
+    switch (label) {
+      case "animal": {
+        feature.put("cat", 1.0);
+        feature.put("dog", 2.0);
+        floatFeatures.put("a", feature);
+        break;
+      }
+      case "color": {
+        feature.put("red", 2.0);
+        feature.put("black", 4.0);
+        floatFeatures.put("c", feature);
+        break;
+      }
+      case "fruit": {
+        feature.put("apple", 1.0);
+        feature.put("kiwi", 3.0);
```

Comment:
how about test feature.put("apple", 1.0); and feature.put("black", 4.0); together?